UnityProject/Assets/Scripts/Items/Weapons/Gun.cs

```diff
@@ -347,24 +369,132 @@ public virtual void ServerPerformInteraction(InventoryApply interaction)
 					if (mag)
 					{
 						ServerHandleReloadRequest(mag.gameObject);
+						return;
 					}
-					else if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Suppressor) &&
-					         !isSuppressed && isSuppressible)
+
+					if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.WeaponAttachable))
 					{
-						SyncIsSuppressed(isSuppressed, true);
-						Inventory.ServerTransfer(interaction.FromSlot, suppressorSlot);
+						if (interaction.UsedObject.TryGetComponent<ItemActionButton>(out var itemActionButton))
+						{
+							if (Inventory.ServerTransfer(interaction.FromSlot, flashlightSlot))
+							{
+								//TODO: itemActionButton for flashlight?
+							}
+							return;
+						}
+
+						if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Suppressor) &&
+					         !isSuppressed && isSuppressible)
+						{
+							if (Inventory.ServerTransfer(interaction.FromSlot, suppressorSlot))
+							{
+								SyncIsSuppressed(isSuppressed, true);
+							}
+							return;
+						}
+
+						if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Knife) &&
```

Comment:
This implementation is going to get messy the more we add more attachments for guns.
It would be better if we had a more generic `GunAttachment` implementation that automatically sets those things up based on the attachment's behaviors.

Reply:
I can add probably add a generic gunattachment implementation, however all of the itemstorage stuff would still need to be changed in the instance of adding a new attachment, also possibly the rightclick menu stuff(?)

Reply:
You can give every gun an attachment config, so when it starts; it reserves the item slots it needs for specific attachments to use.
And the right click can check each attachment if it has a specific interaction it wants to be implemented, and show it to the player.